engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java

```diff
@@ -213,6 +239,44 @@ public void parseSignature() {
         }
     }
 
+    private boolean isSafelyCastable(Set<Class<?>> types, Class<?> type) {
+        for (Class<?> t : types) {
+            if (t.isAssignableFrom(type)) {
+                return true;
+            }
+            if (t.isPrimitive() && type.isPrimitive() && isLosslessWideningPrimitiveConversion(type, t)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public void verifyArguments(Class<?>[] argTypes) {
+        String callableName = pyCallable.getAttribute("__name__").toString();
+
+        for (int i = 0; i < argTypes.length; i++) {
+            Set<Class<?>> types =
+                    parameters.get(i > parameters.size() - 1 ? parameters.size() - 1 : i).getPossibleTypes();
+
+            // to prevent the unpacking of an array column when calling a Python function, we prefix the column accessor
+            // with a cast to generic Object type, until we can find a way to convey that info, we'll just skip the
+            // check for Object type input
+            if (argTypes[i] == Object.class) {
+                continue;
+            }
+
+            Class<?> t = getUnboxedType(argTypes[i]) == null ? argTypes[i] : getUnboxedType(argTypes[i]);
+            if (!types.contains(t) && !types.contains(Object.class)
```

Comment:
Do you need a `PyObject` check?

Reply:
No, No special handling of PyObject. 
1. It can't be used to type hint a parameter in a UDF
2. if it is a type of a column argument, the receiving parameter must be of Py/numpy object type. The same as other  unsupported Java types.